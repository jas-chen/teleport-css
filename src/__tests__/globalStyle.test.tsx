import { expect, test } from 'vitest';
import { renderToStaticMarkup } from 'react-dom/server';
import fnv1a from '@sindresorhus/fnv1a';
import { create } from '..';
import pretty from 'pretty';

function hashFn(value: string) {
  return fnv1a(value, { size: 64 }).toString(36).slice(0, 8);
}

const { renderGlobalStyle } = create({
  hashFn,
});

test('can define animation with objects', () => {
  const globalStyle = renderGlobalStyle(() => ({
    ':root': {
      '--black': '#000',
    },
    body: {
      main: {
        color: 'var(--black)',
      },
    },
  }));

  expect(pretty(renderToStaticMarkup(globalStyle))).toMatchSnapshot();
});

test('can define animation with arrays', () => {
  const globalStyle = renderGlobalStyle(() => [
    {
      ':root': {
        '--black': '#000',
      },
    },
    {
      body: {
        main: {
          color: 'var(--black)',
        },
      },
    },
  ]);

  expect(pretty(renderToStaticMarkup(globalStyle))).toMatchSnapshot();
});

test('no css', () => {
  const globalStyle = renderGlobalStyle(() => null);

  expect(pretty(renderToStaticMarkup(globalStyle))).toMatchSnapshot();
});
