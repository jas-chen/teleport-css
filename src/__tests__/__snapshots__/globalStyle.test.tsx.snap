// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can define animation with arrays 1`] = `
"<style data-precedence="-1" data-href="1f5f73dq">
  :root {
    --black: #000
  }

  body {
    main {
      color: var(--black)
    }
  }
</style>
<style data-precedence="0" data-href="$0"></style>
<style data-precedence="1" data-href="$1"></style>"
`;

exports[`can define animation with objects 1`] = `
"<style data-precedence="-1" data-href="1f5f73dq">
  :root {
    --black: #000
  }

  body {
    main {
      color: var(--black)
    }
  }
</style>
<style data-precedence="0" data-href="$0"></style>
<style data-precedence="1" data-href="$1"></style>"
`;

exports[`no css 1`] = `""`;
